### DISCUSSION TITLE
DAI 2024-2025 - Practical work 1 - Rothen Evan, Thiebaud Jonathan

### DISCUSSION THREAD

**Discussion by JONATHAN-THIEBAUD** [heig-vd-dai-course/heig-vd-dai-course#412] [Show and tell]
Notre idée est de reprendre une modélisation UML que nous avons déjà et de créer l'interface WEB complète en java.

**Comment by ludelafo** [comment]
Bonjour,

Nous ne sommes pas sûrs de comprendre comment cela peut se retranscrire en outil CLI. Est-ce bien dans le scope du projet par rapport à la consigne du travail pratique 1 ?

Volontiers si vous pouvez nous donner des exemples si vous souhaitez garder cette idée.

**Reply by JONATHAN-THIEBAUD** [reply]
Bonjour, 

Nous en avons discuté et nous pensons finalement faire un outil en ligne de commande qui permet de gérer une base de données sur un modèle UML donné. 

Exemples d'opérations : 
CRUD pour les entités du modèle.
Import/export de données depuis/vers des fichiers JSON ou CSV.
Requêtes de recherche sur les relations. 

Exemples de commandes : 
cli add-entity : ajouter une nouvelle entrée.
cli update-entity : mettre à jour une entrée existante.
cli delete-entity :  supprimer une entrée.
cli list-entities : afficher les entrées (avec ou sans filtrage en fonction de la complexité)

Cela paraît réalisable ?

**Reply by ludelafo** [reply]
Bonjour,

J'ai le ressenti que c'est peut-être beaucoup au vu du temps à disposition qu'il vous reste. Un futur travail pratique devrait vous permettre de pouvoir aller dans cette direction avec plus de facilité que ce qui pourrait être fait dans le contexte de ce premier travail pratique.

Si je reprends un des points mentionnés, même si ce n'est pas directement lié à votre idée, une conversion de fichiers JSON <> CSV ou encore XML me parait plus accessible pour ce premier travail pratique.

@hadrylouis, qu'est-ce que tu en penses ?

**Comment by JONATHAN-THIEBAUD** [comment]
Nous pouvons partir la dessus avec une architecture propre avec maven et Picocli , ce sera peut-être plus réaliste pour l'instant. 

## Fonctionnalités
**Conversion JSON vers CSV ou XML** 
**Conversion CSV vers JSON ou XML** 
**Conversion XML vers JSON ou CSV** 

On penses utiliser des classes Java natives pour la lecture et l'écriture de fichiers et la gestion des flux. 

Avec picocli on pourrait gérer les arguments, tels que : 
format source : --input-format
format destination : --output-format
fichiers d'entrée/sortie

On pourrait implémenter quelques tests et logs. 

Cela paraît ok en terme de complexité ? Ou trop simple/complexe ?

**Reply by ludelafo** [reply]
Ça me semble très bien, merci beaucoup !

**Reply by JONATHAN-THIEBAUD** [reply]
Parfait.

**Comment by JONATHAN-THIEBAUD** [comment]
# Submission of the practical work 1 : 
**link to the repo :** https://github.com/JONATHAN-THIEBAUD/FileConverter
**Last commit hash :** 4662085412ea4cf53c7bc03bac4a63b0ffb0d5d3
